### PR TITLE
fix get_thread_num bug

### PR DESCRIPTION
**Pull request by BradSwain**
The visited cache stored functions, even though the `computeGuardedBlocks` only computes guarded blocks for a single call.

This caused a bug where if functions had more than one block guarded by `omp_get_thread_num` the second one would be ignored.

### CODE REVIEW

**Review by funemy — COMMENTED** [review]
I fail to see any actual changes other than renaming?

What does this change affect?

File: src/Analysis/OpenMPAnalysis.cpp [hunk 1]
```diff
@@ -836,12 +836,12 @@ std::set<const llvm::BasicBlock *> getGuardedBlocks(const llvm::BranchInst *bran
 
 void SimpleGetThreadNumAnalysis::computeGuardedBlocks(const Event *event) {
   assert(event->getIRType() == IR::Type::OpenMPGetThreadNum);
-  auto const func = event->getFunction();
+  auto const call = event->getInst();
```
Comment (funemy) [inline src/Analysis/OpenMPAnalysis.cpp:839]: Oh, I see.

Previously you only cache by `Function`, now you cache by `Instruction`.
Is that the purpose of this PR?

**Comment by BradSwain** [comment]
The `visited` structure is used to prevent us from redoing computation.

https://github.com/coderrect-inc/OpenRace/blob/b3edc15b90e108b7c751abff0aaee78bde2e715c/src/Analysis/OpenMPAnalysis.cpp#L840-L841

Before the visited structure was saving functions, even though we only compute guarded blocks for a single `omp_get_thread_num` call. 
So if that function had more than one `omp_get_thread_num` call the guarded blocks would only be computed for the first call.
The later calls would see that the function had already been visited and return early form the `computeGuardedBlocks` function.


Now the `visited` structure saves a call instruction. So we compute guarded blocks once for each call to `omp_get_thread_num`.

File: src/Analysis/OpenMPAnalysis.cpp (hunk 1 above)
Comment (BradSwain) [reply src/Analysis/OpenMPAnalysis.cpp:839]: Accidentally answered as a new comment: https://github.com/coderrect-inc/OpenRace/pull/253#issuecomment-877474575

Comment (BradSwain) [reply src/Analysis/OpenMPAnalysis.cpp:839]: yes

File: tests/data/integration/openmp/get-thread-num-double-no.c [hunk 2]
```diff
@@ -0,0 +1,21 @@
+#include <omp.h>
+#include <stdio.h>
+
+int main() {
+  int counter = 0;
+
+#pragma omp parallel
+  {
+    int tid = omp_get_thread_num();
```
Comment (funemy) [inline tests/data/integration/openmp/get-thread-num-double-no.c:9]: But this example still has only one call to `omp_get_thread_num`.

Shouldn't it have at least 2 calls in order to test your PR?

Comment (BradSwain) [reply tests/data/integration/openmp/get-thread-num-double-no.c:9]: good catch, lol.

Comment (BradSwain) [reply tests/data/integration/openmp/get-thread-num-double-no.c:9]: Me writing tests like that is how this bug was introduced in the first place lol

**Review by funemy — APPROVED** [review]

**Pull request merged**